### PR TITLE
Add support for Visual Studio 2015 & 2017

### PR DESCRIPTION
**Pull request by djipi**
The stdint.h file contains the Visual Studio 2015 & 2017 detection changes.
The Visual Studio 2015 & 2017 projects can probably be put aside but I let you to decide.

### CODE REVIEW

**Comment by twinaphex** [comment]
Umm, please implement these as Makefile targets instead. We want to have to maintain zero Visual Studio solution files, and there are ways to get this compiling properly with a Makefile.

**Comment by twinaphex** [comment]
I can ask @bparker06 if he can do this at least for you since I understand it might not be a reasonable expectation to expect you to dig through all those files. We'll see what happens.

**Comment by djipi** [comment]
I understand, please let me know how @bparker06 feels about it. Otherwise, I will try to modify the makefile targets.
Just for my curiosity, could you also tell me the reason(s) behind the zero Visual Studio solution files?

**Comment by Alcaro** [comment]
Because adding those two solutions means the procedure for adding a new source file changes from "create file, add to Makefile" to "create file, add to Makefile, Win-VS2015.vcxproj, Win-VS2015.vcxproj.filters, Win-VS2017.vcxproj and Win-VS2017.vcxproj.filters". We'd rather not deal with that maintenance burden. Especially when the build works for most of us if we forget the solutions.

If you can find a way to do wildcards in a VS project, I believe we'll have no objections.

(Obvious counterargument: add file to CMakelists.txt instead. We've considered and rejected that option too, but I don't remember the relevant argument.)

**Comment by twinaphex** [comment]
Hi there @djipi ,

I believe @bparker06 is going to add VS2015/2017 targets himself, so let's give him some time to get it done. After it is done, we can reconvene and you can tell us if you can get it easily compiling, and if so, we can close this issue. I will leave it open until then.

**Comment by twinaphex** [comment]
Hi there,

bparker added this -

https://github.com/libretro/virtualjaguar-libretro/commit/2b194953d35ab0a264cbdbea8a82db0b03c27a48

It should now be possible to compile with MSVC like this -

MSVC 2015 desktop x86

make -f Makefile platform=windows_msvc2015_x86

MSVC 2015 desktop x64

make -f Makefile platform=windows_msvc0215_x64

MSVC 2017 desktop x86

make -f Makefile platform=windows_msvc2017_desktop_x86

MSVC 2017 desktop x64

make -f Makefile platform=windows_msvc2017_desktop_x64

MSVC 2017 UWP x86

make -f Makefile platform=windows_msvc2017_uwp_x86

MSVC 2017 UWP x64

make -f Makefile platform=windows_msvc2017_uwp_x64

Let me know if this works for you @djipi. You might need MSYS2 installed in order to use make but that should be it.

**Comment by djipi** [comment]
Thank you for the modifications.
I have tested the "MSVC 2017 desktop x64" under cygwin64 and it went well. I will check the another possibilities later.
I have kept my changes in the stdint.h file, so I guess you will have to merge my line if you want to test the makefile at your side.
Thanks.

**Comment by twinaphex** [comment]
OK, tht stdint.h change seems like a good improvement. I will backport that, and I'll also backport it to all the other libretro-common-related repos (RetroArch/libretro-common, etc.)

File: libretro-common/include/compat/msvc/stdint.h [hunk 1]
```diff
@@ -31,8 +31,8 @@
 #ifndef __RARCH_STDINT_H
 #define __RARCH_STDINT_H
 
-#if _MSC_VER && (_MSC_VER < 1600)
```
Comment (twinaphex) [inline libretro-common/include/compat/msvc/stdint.h]: Hmm, still uncertain why this change is needed though.

This seems to indicate MSVC has had stdint.h since 2010.

Comment (djipi) [reply libretro-common/include/compat/msvc/stdint.h]: The 1600 value is for Visual Studio 2010 (MSVC++ 10.0), if you want to support 2017 (MSVC++ 14.1) and below, the value needs to be 1911 otherwise compiler errors will occur.

Comment (twinaphex) [reply libretro-common/include/compat/msvc/stdint.h]: @bparker06 What do you make of this change?

Comment (djipi) [reply libretro-common/include/compat/msvc/stdint.h]: I'm not sure to understand the question.
This information comes from a list of values depend the Visual Studio versions:

MSVC++ 4.x  _MSC_VER == 1000
MSVC++ 5.0  _MSC_VER == 1100 (Visual Studio 97)
MSVC++ 6.0  _MSC_VER == 1200 (Visual Studio 6.0)
MSVC++ 7.0  _MSC_VER == 1300 (Visual Studio .NET 2002)
MSVC++ 7.1  _MSC_VER == 1310 (Visual Studio .NET 2003)
MSVC++ 8.0  _MSC_VER == 1400 (Visual Studio 2005)
MSVC++ 9.0  _MSC_VER == 1500 (Visual Studio 2008)
MSVC++ 10.0 _MSC_VER == 1600 (Visual Studio 2010)
MSVC++ 11.0 _MSC_VER == 1700 (Visual Studio 2012)
MSVC++ 12.0 _MSC_VER == 1800 (Visual Studio 2013)
MSVC++ 14.0 _MSC_VER == 1900 (Visual Studio 2015)
MSVC++ 14.1 _MSC_VER >= 1910 (Visual Studio 2017)

Comment (bparker06) [reply libretro-common/include/compat/msvc/stdint.h]: Not sure what is trying to be accomplished here... I was able to compile for VC2015 and 2017 with the current HEAD without applying any patches from this PR... we use our current stdint.h with every compiler version from VC6 all the way up to 2017.

Comment (djipi) [reply libretro-common/include/compat/msvc/stdint.h]: That's fine, so you can discard my modification then.
At my side, I had a compilation issue; I will try again to compile without the modification, and if I got the error, I will simply keep my modification locally.
I'm sincerely sorry to have bothered you with this.

Comment (twinaphex) [reply libretro-common/include/compat/msvc/stdint.h]: No need to apologize djipi, in fact we should thank you instead for paving the road towards MSVC2017 support like this.
